app/src/js/reducers/collections.js

```diff
@@ -1,156 +1,106 @@
 'use strict';
-import { set, del } from 'object-path';
-import assignDate from './assign-date';
 
+import { createReducer } from '@reduxjs/toolkit';
+import { deconstructCollectionId } from '../utils/format';
+import assignDate from './utils/assign-date';
 import {
-  COLLECTION,
-  COLLECTION_INFLIGHT,
-  COLLECTION_ERROR,
-  COLLECTION_APPLYWORKFLOW,
-  COLLECTION_APPLYWORKFLOW_INFLIGHT,
+  createClearItemReducer,
+  createErrorReducer,
+  createInflightReducer,
+  createSuccessReducer
+} from './utils/reducer-creators';
+import {
+  CLEAR_COLLECTIONS_FILTER,
+  CLEAR_COLLECTIONS_SEARCH,
   COLLECTION_APPLYWORKFLOW_ERROR,
-  COLLECTIONS,
-  COLLECTIONS_INFLIGHT,
+  COLLECTION_APPLYWORKFLOW_INFLIGHT,
+  COLLECTION_APPLYWORKFLOW,
+  COLLECTION_DELETE_ERROR,
+  COLLECTION_DELETE_INFLIGHT,
+  COLLECTION_DELETE,
+  COLLECTION_ERROR,
+  COLLECTION_INFLIGHT,
+  COLLECTION,
   COLLECTIONS_ERROR,
-  NEW_COLLECTION,
-  NEW_COLLECTION_INFLIGHT,
+  COLLECTIONS_INFLIGHT,
+  COLLECTIONS,
+  FILTER_COLLECTIONS,
   NEW_COLLECTION_ERROR,
-  COLLECTION_DELETE,
-  COLLECTION_DELETE_INFLIGHT,
-  COLLECTION_DELETE_ERROR,
-  UPDATE_COLLECTION,
-  UPDATE_COLLECTION_INFLIGHT,
-  UPDATE_COLLECTION_ERROR,
-  UPDATE_COLLECTION_CLEAR,
+  NEW_COLLECTION_INFLIGHT,
+  NEW_COLLECTION,
   SEARCH_COLLECTIONS,
-  CLEAR_COLLECTIONS_SEARCH,
-  FILTER_COLLECTIONS,
-  CLEAR_COLLECTIONS_FILTER
+  UPDATE_COLLECTION_CLEAR,
+  UPDATE_COLLECTION_ERROR,
+  UPDATE_COLLECTION_INFLIGHT,
+  UPDATE_COLLECTION,
 } from '../actions/types';
-import { createReducer } from '@reduxjs/toolkit';
-import { deconstructCollectionId } from '../utils/format';
 
 export const initialState = {
   list: {
     data: [],
     meta: {},
-    params: {}
+    params: {},
   },
   created: {},
   deleted: {},
   executed: {},
   map: {},
-  updated: {}
+  updated: {},
 };
 
 export default createReducer(initialState, {
-  [COLLECTION]: (state, action) => {
-    const { id, data } = action;
+  [COLLECTION]: ({ deleted, map }, { id, data }) => {
     const { name } = deconstructCollectionId(id);
     const collection = data.results.find((element) => element.name === name);
-    set(state, ['map', id, 'inflight'], false);
```

Comment:
what happened to this line?
`set(state, ['map', id, 'inflight'], false);`

Reply:
No, the `inflight` property is _removed_, rather than set to `false`, which achieves the same effect.

Reply:
I restored this explicit setting of `inflight` to `false`